### ISSUE TITLE
Error while using tcp transport

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#34757]
### Description of Issue/Question
When adding `transport: tcp` in the master and minion config I am receivng the following error when running commands against the minion: `Future exception was never retrieved: StreamClosedError`

### Setup
1. setup a master/minion (i have tested this on both ubuntu14 and cenots7 with same results)
2. add `transport: tcp` to the master and minion config
3. restart master and minion

For clarity here are my configs:

```
[root@li1189-103 ~]# grep -E -v '^$|#' /etc/salt/master 
log_level: debug
keep_jobs: 1 
transport: tcp
[root@li1189-103 ~]# grep -E -v '^$|#' /etc/salt/minion
master: localhost
log_level: debug
transport: tcp
```

### Steps to Reproduce Issue
1. 

```
[root@li1189-103 ~]# salt '*' test.ping
[ERROR   ] Future exception was never retrieved: StreamClosedError
li1189-103.members.linode.com:
    True
```

```
[root@li1189-103 ~]# salt-call test.ping
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] The `lspci` binary is not available on the system. GPU grains will not be available.
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506')
[DEBUG   ] Initializing new AsyncTCPReqChannel for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506', 'clear')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] tcp stream to 127.0.0.1:4506 closed, unable to recv
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncTCPReqChannel for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[ERROR   ] Future exception was never retrieved: StreamClosedError
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded test.ping
[DEBUG   ] Initializing new AsyncTCPReqChannel for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'li1189-103.members.linode.com', 'tcp://127.0.0.1:4506')
[ERROR   ] Future exception was never retrieved: StreamClosedError
[DEBUG   ] LazyLoaded nested.output
local:
    True
```

### Versions Report

```
[root@li1189-103 ~]# salt --versions-report
Salt Version:
           Salt: 2015.8.11
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 4.5.5-x86_64-linode69
         system: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
Also my futures version on centos: `python-futures-3.0.3-1.el7.noarch` 
on ubuntu14:  `futures (3.0.5)`

**Comment by thatch45** [comment]
I can't reproduce this on 2016.3, and since there were major additions to the tcp transport between 2015.8 and 2016.3 I am going to say that this has been resolved.
If you disagree let me know :)
otherwise please feel free to close this out as fixed in 2016.3.1, thanks @Ch3LL
